### DISCUSSION TITLE
custom configure logger

### DISCUSSION THREAD

**Discussion by joebnb** [verdaccio/verdaccio#2842] [Q&A]
pino not work well with pm2,could revert back previous logger or make user can custome it ?

i think revert is a good choice cause they(pino) didn't have a plan to fix them in past 3 years,i think it wont fix in further

**Comment by juanpicado** [comment]
I think the root issue is on pm2 rather on pino https://github.com/pinojs/pino/issues/491#issuecomment-639538241 have you consider find another alternative to pm2?

**Reply by joebnb** [reply]
have you consider find another alternative to pm2?
------
no,it's seems pm2 in process guard  field good than others,especially cluster mode and load banlance for node.js app

i use pm2 to start verdaccio with 8 worker,verdaccio will have a big improve in multi request

**Reply by joebnb** [reply]
hi,@juanpicado did you see this https://github.com/pinojs/pino/issues/491#issuecomment-639483661 https://github.com/pinojs/pino/issues/491#issuecomment-639485364 https://github.com/pinojs/pino/issues/491#issuecomment-639485919 ,it's seems mean if remove destination in logger create,pino will work with pm2,so did we can remove destination when log type is stdout

**Reply by juanpicado** [reply]
I really want to keep it asyncronous by default, decrease the performance by removing the default destination https://github.com/pinojs/pino/issues/491#issuecomment-639538241 is a huge trade-off I'd not want to pay. I agree with this comment https://github.com/pinojs/pino/issues/491#issuecomment-639601400

What we can do is allow you to remove the destination by configuration `config.yaml` and then is up to you deal with performance, but https://github.com/verdaccio/verdaccio/pull/2898 does decrease the performance of the registry in all inveronments, this is a no go for me.

eg: https://verdaccio.org/docs/logger#configuration <- maybe add some configuration as type that allows that, for instance.

````
logs: { type: stdout-synchronous, format: pretty, level: http }
````
